### PR TITLE
Don't crash if a framegraph attachment is unused

### PR DESCRIPTION
**Pull request by kh-huawei**
## What does this PR do?

Adds a few checks to avoid an exception when an attachment is unused. 

## How was this PR tested?

Add an image-attachment without any connections to the Parent-Pass in `MainPipeline.pass`. The result is now the message 
`[Error] (FrameGraph) - Invalid State: attachment 'Root.MainPipeline_0.TestImage' was added but never used!`, but no crash.

### CODE REVIEW

File: Gems/Atom/RHI/Code/Source/RHI/FrameGraphCompiler.cpp [hunk 1]
```diff
@@ -284,6 +284,11 @@ namespace AZ::RHI
         for (FrameAttachment* transientImage : attachmentDatabase.GetTransientImageAttachments())
         {
             Scope* firstScope = transientImage->GetFirstScope();
+            if (firstScope == nullptr)
+            {
+                // The attachment is unused: We will get a warning in ValidateEnd(), but we don't want to crash here
```
Comment (moudgils) [inline Gems/Atom/RHI/Code/Source/RHI/FrameGraphCompiler.cpp:289]: Can you expand on the comment as to what type of use case can cause the first scope of a transient resource to be null.

File: Gems/Atom/RHI/Code/Source/RHI/FrameGraphCompiler.cpp [hunk 2]
```diff
@@ -547,8 +552,15 @@ namespace AZ::RHI
             for (uint32_t attachmentIndex = 0; attachmentIndex < (uint32_t)transientBufferGraphAttachments.size(); ++attachmentIndex)
             {
                 BufferFrameAttachment* transientBuffer = transientBufferGraphAttachments[attachmentIndex];
-                const uint32_t scopeIndexFirst = transientBuffer->GetFirstScope()->GetIndex();
-                const uint32_t scopeIndexLast = transientBuffer->GetLastScope()->GetIndex();
+                const auto* firstScope = transientBuffer->GetFirstScope();
+                const auto* lastScope = transientBuffer->GetLastScope();
+                if (firstScope == nullptr || lastScope == nullptr)
+                {
+                    // The attachment is unused: We will get a warning in ValidateEnd(), but we don't want to crash here
```
Comment (moudgils) [inline Gems/Atom/RHI/Code/Source/RHI/FrameGraphCompiler.cpp:559]: Same

File: Gems/Atom/RHI/Code/Source/RHI/FrameGraphCompiler.cpp [hunk 3]
```diff
@@ -557,8 +569,15 @@ namespace AZ::RHI
             for (uint32_t attachmentIndex = 0; attachmentIndex < (uint32_t)transientImageGraphAttachments.size(); ++attachmentIndex)
             {
                 ImageFrameAttachment* transientImage = transientImageGraphAttachments[attachmentIndex];
-                const uint32_t scopeIndexFirst = transientImage->GetFirstScope()->GetIndex();
-                const uint32_t scopeIndexLast = transientImage->GetLastScope()->GetIndex();
+                const auto* firstScope = transientImage->GetFirstScope();
+                const auto* lastScope = transientImage->GetLastScope();
+                if (firstScope == nullptr || lastScope == nullptr)
+                {
+                        // The attachment is unused: We will get a warning in ValidateEnd(), but we don't want to crash here
```
Comment (moudgils) [inline Gems/Atom/RHI/Code/Source/RHI/FrameGraphCompiler.cpp:576]: Same

File: Gems/Atom/RHI/Code/Source/RHI/FrameGraphCompiler.cpp (hunk 1 above)
Comment (kh-huawei) [reply Gems/Atom/RHI/Code/Source/RHI/FrameGraphCompiler.cpp]: expanded the comment

File: Gems/Atom/RHI/Code/Source/RHI/FrameGraphCompiler.cpp (hunk 2 above)
Comment (kh-huawei) [reply Gems/Atom/RHI/Code/Source/RHI/FrameGraphCompiler.cpp]: same

File: Gems/Atom/RHI/Code/Source/RHI/FrameGraphCompiler.cpp (hunk 3 above)
Comment (kh-huawei) [reply Gems/Atom/RHI/Code/Source/RHI/FrameGraphCompiler.cpp]: same

**Review by jhmueller-huawei — APPROVED** [review]

**Review by moudgils — APPROVED** [review]

**Pull request merged**